src/Compiler/Checking/CheckExpressions.fs

```diff
@@ -5292,6 +5292,11 @@ and TcExprsWithFlexes (cenv: cenv) env m tpenv flexes argTys args =
     (tpenv, List.zip3 flexes argTys args) ||> List.mapFold (fun tpenv (flex, ty, e) ->
          TcExprFlex cenv flex false ty env tpenv e)
 
+and TcExprsNoFlexes (cenv: cenv) env m tpenv argTys args =
+    if List.length args <> List.length argTys then error(Error(FSComp.SR.tcExpressionCountMisMatch((List.length argTys), (List.length args)), m))
```

Comment:
Minor nit: we mostly use `.Length` property in the compiler. Add a type annotation if needed.